Feature/clean up tests

soyoboyo opened the pull request.

SpokoSpartan committed:
Change all tests to use AssertAll()

SpokoSpartan committed:
Add descriptions for almost all tests

soyoboyo committed:
Merge develop into feature/clean_up_tests

ghost commented:
## DeepCode's analysis on [#2e02f8](https://www.deepcode.ai/app/gh/tubidubidam/tim2/3b7e696fbd55a35bc0c6a443c12cf3097aac1667/tubidubidam/tim2/2e02f886661d1b3cbb48ad819f61eb6d84efbaeb/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=3&) found:

- :warning: **1** warning, :information_source: **3** minor issues. :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">The value assigned at = is overwritten at a following statement. Occurrences: <ul><li><a href="https://github.com/tubidubidam/tim2/blob/2e02f886661d1b3cbb48ad819f61eb6d84efbaeb/frontend/src/app/features/developers/dev-messages/dev-messages-table/dev-messages-table.component.ts#L65">dev-messages-table.component.ts:65</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/tubidubidam/tim2/3b7e696fbd55a35bc0c6a443c12cf3097aac1667/tubidubidam/tim2/2e02f886661d1b3cbb48ad819f61eb6d84efbaeb/pr/_/%2Ffrontend%2Fsrc%2Fapp%2Ffeatures%2Fdevelopers%2Fdev-messages%2Fdev-messages-table%2Fdev-messages-table.component.ts/javascript%2Fdc%2FOverwriteAssignment/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Use java.util.ArrayList instead of java.util.LinkedList. Using java.util.LinkedList can lead to unnecessary performance overhead if the list is randomly accessed. Occurrences: <ul><li><a href="https://github.com/tubidubidam/tim2/blob/2e02f886661d1b3cbb48ad819f61eb6d84efbaeb/backend/src/test/java/org/tim/services/ReportServiceTests.java#L177">ReportServiceTests.java:177</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/tubidubidam/tim2/3b7e696fbd55a35bc0c6a443c12cf3097aac1667/tubidubidam/tim2/2e02f886661d1b3cbb48ad819f61eb6d84efbaeb/pr/_/%2Fbackend%2Fsrc%2Ftest%2Fjava%2Forg%2Ftim%2Fservices%2FReportServiceTests.java/java%2Fdc%2FApiMigration~java.util.LinkedList%2Ftest/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Use java.io.FileOutputStream instead of java.io.FileWriter. Occurrences: <ul><li><a href="https://github.com/tubidubidam/tim2/blob/2e02f886661d1b3cbb48ad819f61eb6d84efbaeb/backend/src/main/java/org/tim/services/ReportService.java#L124">ReportService.java:124</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/tubidubidam/tim2/3b7e696fbd55a35bc0c6a443c12cf3097aac1667/tubidubidam/tim2/2e02f886661d1b3cbb48ad819f61eb6d84efbaeb/pr/_/%2Fbackend%2Fsrc%2Fmain%2Fjava%2Forg%2Ftim%2Fservices%2FReportService.java/java%2Fdc%2FApiMigration~java.io.FileWriter/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/tubidubidam/tim2/3b7e696fbd55a35bc0c6a443c12cf3097aac1667/tubidubidam/tim2/2e02f886661d1b3cbb48ad819f61eb6d84efbaeb/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=3&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=tubidubidam)

soyoboyo committed:
# Conflicts:
#	backend/pom.xml
#	backend/src/main/java/org/tim/configurations/securityConfig/ResourceServerConfig.java
#	backend/src/main/java/org/tim/constants/CSVFileConstants.java
#	backend/src/main/java/org/tim/constants/Mappings.java
#	backend/src/main/java/org/tim/constants/TranslationStatus.java
#	backend/src/main/java/org/tim/constants/UserMessages.java
#	backend/src/main/java/org/tim/controllers/ProjectController.java
#	backend/src/main/java/org/tim/controllers/ReportController.java
#	backend/src/main/java/org/tim/controllers/UserController.java
#	backend/src/main/java/org/tim/databaseSeed/MessageSeeder.java
#	backend/src/main/java/org/tim/databaseSeed/ProjectSeeder.java
#	backend/src/main/java/org/tim/entities/Project.java
#	backend/src/main/java/org/tim/entities/ReportDataRow.java
#	backend/src/main/java/org/tim/services/ReportService.java
#	backend/src/test/java/org/tim/controllers/ProjectControllerTestIT.java
#	backend/src/test/java/org/tim/controllers/TranslationControllerTestIT.java
#	backend/src/test/java/org/tim/services/ReportServiceTests.java
#	frontend/src/app/features/developers/aggregate-info/aggregate-info.component.ts
#	frontend/src/app/material.module.ts
